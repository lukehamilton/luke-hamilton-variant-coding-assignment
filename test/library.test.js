import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, configure } from 'enzyme'

import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

import DebouncedInput from '../shared/library/inputs/debounced'

describe('DebouncedInput', () => {
  test('it renders correctly', () => {
    expect(renderer.create(<DebouncedInput />).toJSON()).toMatchSnapshot()
  })
  test('it triggers onKeyDown', () => {
    const onKeyDown = jest.fn()
    const el = shallow(<DebouncedInput onKeyDown={onKeyDown} />, {
      disableLifecycleMethods: true
    })
    el.simulate('keydown')
    const calls = onKeyDown.mock.calls
    expect(calls.length).toBe(1)
  })
})
