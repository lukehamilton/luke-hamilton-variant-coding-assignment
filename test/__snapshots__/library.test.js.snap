// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DebouncedInput it renders correctly 1`] = `
<input
  className="css-ox8hf7-Input e1i9eloy0"
  onChange={[Function]}
/>
`;
